[DevOps]: Fix latest `mypy` pre-commit errors on `main`

https://github.com/xCDAT/xcdat/actions/runs/7390850971/job/20106586822

```python
 pyproject.toml: [mypy]: python_version: Python 3.1 is not supported (must be 3.4 or higher). You may need to put quotes around your Python version
xcdat/temporal.py:735: error: Argument 1 to "drop_vars" of "Dataset" has incompatible type "Hashable"; expected "str | Iterable[Hashable] | Callable[[Dataset], str | Iterable[Hashable]]"  [arg-type]
Found 1 error in 1 file (checked 25 source files)
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -40,7 +40,7 @@ markers = ["flaky", "network"]
 
 [tool.mypy]
 # Docs: https://mypy.readthedocs.io/en/stable/config_file.html
-python_version = 3.10
+python_version = "3.10"
 check_untyped_defs = true
 ignore_missing_imports = true
 warn_unused_ignores = true

--- a/xcdat/temporal.py
+++ b/xcdat/temporal.py
@@ -732,7 +732,7 @@ class TemporalAccessor:
             # The original time dimension name is restored after grouped
             # arithmetic, so the labeled time dimension name is no longer needed
             # and therefore dropped.
-            ds_obs = ds_obs.drop_vars(self._labeled_time.name)
+            ds_obs = ds_obs.drop_vars(str(self._labeled_time.name))
 
         if weighted and keep_weights:
             self._weights = ds_climo.time_wts

